source/frontend/src/components/maps/leaflet/Markers/SingleMarker.tsx

```diff
@@ -37,22 +50,30 @@ const SinglePropertyMarker: React.FC<React.PropsWithChildren<SinglePropertyMarke
     const [longitude, latitude] = pointFeature.geometry.coordinates;
 
     const latlng = { lat: latitude, lng: longitude };
-    if (isOwned) {
+    if (isPimsLocation(pointFeature)) {
       mapMachine.mapMarkerClick({
         clusterId: clusterId,
         latlng: latlng,
-        pimsFeature: null,
-        fullyAttributedFeature: (
-          pointFeature as PointFeature<PMBC_FullyAttributed_Feature_Properties>
-        ).properties,
+        pimsLocationFeature: pointFeature.properties,
+        pimsBoundaryFeature: null,
+        fullyAttributedFeature: null,
       });
-    } else {
+    } else if (isPimsBoundary(pointFeature)) {
       mapMachine.mapMarkerClick({
         clusterId: clusterId,
```

Comment:
do boundaries have cluster ids?

Reply:
they should, all features (should) have a cluster id